Indicators/RelativeVigorIndex.cs

```diff
@@ -0,0 +1,133 @@
+            RviWindow = new RollingWindow<IndicatorDataPoint>(period);
+            Bars = new RollingWindow<TradeBar>(period);
+        }
+
+
+        /// <summary>
+        /// Automatically names the indicator
+        /// </summary>
+        /// <param name="period">int - the length of computation cycle</param>
+        public RelativeVigorIndex(int period)
+            : this("RVI_" + period, period)
+        {
+        }
+
+        /// <summary>
+        /// Gets a flag indicating when this indicator is ready and fully initialized
+        /// </summary>
+        public override bool IsReady
+        {
+            get { return value1.IsReady && value2.IsReady && RviWindow.IsReady; }
+        }
+        /// <summary>
+        /// Computes the next value of this indicator from the given state
+        /// </summary>
+        /// <param name="input">The input given to the indicator</param>
+        /// <returns>A new value for this indicator</returns>
+        protected override decimal ComputeNextValue(TradeBar input)
+        {
+            // Save the bar for summing
+            Bars.Add(input);
+
+            // check if it is not ready add a 0 to the RviWindow
+            if (!Bars.IsReady)
+            {
+                RviWindow.Add(new IndicatorDataPoint(input.EndTime, 0.0m));
+            }
+            else
+            {
+                // Bars have been added, so Bars[0] is the current value of the input
+
+
+                // Filter the Close - Open with a four-bar symetical FIR filter before the terms are summed.
+                var v1 = ((Bars[0].Close - Bars[0].Open) + 2 * (Bars[1].Close - Bars[1].Open) +
+                          2 * (Bars[2].Close - Bars[3].Open) + (Bars[3].Close - Bars[3].Open) / 6);
+                value1.Add(new IndicatorDataPoint(input.EndTime, v1));
+
+                // Filter the High - Low with a four-bar symetical FIR filter before the terms are summed.
+                var v2 = ((Bars[0].High - Bars[0].Low) + 2 * (Bars[1].High - Bars[1].Low) +
+                          2 * (Bars[2].High - Bars[3].Low) + (Bars[3].High - Bars[3].Low) / 6);
+                value2.Add(new IndicatorDataPoint(input.EndTime, v2));
+
+                // The numerator and denominator are summed independently
+                decimal num = value1.Sum(point => point.Value);
+                decimal denom = value2.Sum(point => point.Value);
+
+                try
+                {
+                    // The RVI is the ratio of the numerator to the denominator.  Since
+                    //  they are lagged by the same amount, due to the filtering,
+                    //  the lag is removed by taking the ratio.
+                    RviWindow.Add(new IndicatorDataPoint(input.EndTime, num / denom));
+                }
+                catch (DivideByZeroException zex)
+                {
+                    throw new Exception(zex.Message + zex.StackTrace);
```

Comment:
catching and rethrowing exceptions adds little and hides the original stack trace. Please remove the try/catch/rethrow new exception.